feat: Discovery 10 Year Anniversary Blog Post (#3007)
agrc/gis.utah.gov: src/content/blog/2025-05-05-celebrating-ten-years-of-discover.mdx

description: >-
  We are celebrating 10 years of the Discover Basemap and Imagery Service this
  month! Discover is our one-stop shop for high-resolution aerial imagery and
  custom Utah basemaps, all available on our website for free and licensed use.
date: '2025-05-05T20:08:26.922Z'
category: Discover
cover_image: /src/images/pillar-blog/default-social-card.png
cover_image_alt: ugrc social card
---

import Contacts from '@components/page/Contacts.astro';

We are celebrating 10 years of the Discover Basemap and Imagery Server this month! [Discover](/products/discover/) is our one-stop shop for aerial imagery and custom Utah basemap services, all available from UGRC for free to public and licensed users. Our nearly 2,000 registered accounts include government agencies such as the Utah Department of Transportation, the Utah Department of Public Safety, the Utah Division of Wildlife Resources, and the Utah Division of Forestry, Fire, and State Lands. We also serve a wide variety of local governments, private firms, K-12 and higher education, and many other organizations that use the services to benefit the lives of everyday citizens in Utah.

## So, where did Discover come from?

> Before we implemented the current Discover service, UGRC utilized a platform called [ArcGIS Image Server]](https://enterprise.arcgis.com/en/image/latest/get-started/windows/what-is-arcgis-image-server-.htm). It was available for public and private use, but connections made to the service created a heavy digital load on the ArcGIS Server. This resulted in a number of problems for both UGRC and our user base; the services experienced rendering latenecy and occasional downtime. While this old service did provide significant customization options for imagery and topographic maps, it was not the best solution.

Reviewer: ```suggestion
Before we implemented the current Discover service, UGRC utilized a platform called [ArcGIS Image Server]](https://enterprise.arcgis.com/en/image/latest/get-started/windows/what-is-arcgis-image-server-.htm). It was available for public and private use, but connections made to the service created a heavy digital load on our ArcGIS Server. This resulted in a number of problems for both UGRC and our user base; the services experienced rendering latenecy and occasional downtime. While this old service did provide significant customization options for imagery and topographic maps, it was not the best solution.
```

Just making it clear that this was a server we had to maintain, not something Esri was providing for us.

For a period of time, UGRC resolved Utah's need for high-resolution aerial imagery through contracts with federal programs. These programs provided a patchwork of imagery every couple of years, but in 2012, those programs came to an end. Thus, we needed a new solution.

## From Giza to Discover

Shortly after high resolution aerial imagery through federal programs was no longer available, UGRC began working with Google and [Sanborn](https://sanborn.com/) to create a tile pyrimid server appliance called Giza, which serves Google imagery and basemaps as WMS and WMTS. UGRC was one of the earliest adopters of this new Google imagery program. We developed a Utah-specific version of Giza and named it Discover. In 2015, Discover officially launched as Utah's premier source for aerial imagery and basemaps services.

> But, as times change, so do the relationships between government entities and private corporations. After a few years of working with Google, we eventually switched to working with [Hexagon](https://hexagon.com/products/aerial-imagery) in 2018. Hexagon provides a statewide high resolution version of the federally administered [National Agriculture Imaging Program (NAIP)](https://naip-usdaonline.hub.arcgis.com/), creates tile pyramids from the imagery, and uploads it to the cloud. From there, our automated service pulls from that cloud storage to provide you with readily available aerial imagery. Since the flights are flown as part of the NAIP, updated imagery is available every three years.

Reviewer 2: we could but I'm not sure what details are important to the audience

> ## Challenges to Overcome

Reviewer 3: My note for this section would be to cut it. I don't think it fits into a celebration post and it reads to me like an LLM wrote it.

Reviewer 2: @miriamseely I'll let you decide on this suggestion

Reviewer 4: I'm fine with taking it out, I don't think I mentioned Hexagon or NAIP elsewhere in the post, but it's possible we don't necessarily need to mention them.

> UGRC's new service worked the first time without a hitch and nothing remotely bad ever happened. There were so few bugs in the system, it may as well have been the pristine, barren landscape of Antarctica. There wasn't even one user complaint…said no developer ever, right? Like any new technology, there were more than a few hurdles to jump over in order to create a streamlined, modernized server for Utah. Many of those hurdles had to do with two unique aspects of Discover: serving the millions of pre-cached cloud-based tiles, and user access.

Reviewer: In the last sentence we call out two problems: tile issues and user access. However, we never mention user access again. Are we missing a paragraph, or should we remove the reference to user access?

Reviewer 2: This originally stated something different. I'll edit it to just be a 'millions of tiles' issue.

> When you view services through Discover, you may think you are looking at a one really big image of Utah. But, this image of Utah is actually composed of millions of image 'tiles', all mosaicked together into a series (just like in a real mosaic floor or ceiling, it takes many colorful tiles to make up the whole artwork). These tiles are pre-cached to increase rendering speed and allows for seamless transitions as you pan or zoom between them. They render so quickly, that many users don't even realize there is more than one tile to begin with.

Reviewer: ```suggestion
When you view services through Discover, you may think you are looking at a one really big image of Utah. But, this image of Utah is actually composed of millions of image 'tiles', all mosaicked together into a series (just like in a real mosaic floor or ceiling, it takes many colorful tiles to make up the whole artwork). These tiles are pre-cached to increase rendering speed and allow for seamless transitions as you pan or zoom between them. They render so quickly that many users don't even realize there is more than one tile to begin with.
```

However, like ragtag siblings stuck in their “get-along” shirt, some of these tiles didn't quite play nice at first. Some of them were so-called 'sticky tiles': when the user zoomed in or out, the other nearby tiles would zoom in or out as well, but sticky tiles would be stuck at the previous zoom level as though glued in place. Other problem children included 'delinquent tiles', which were tiles that didn't show up in the service at all. Apparently, those tiles had somewhere better to be. And many of the other perfectly innocent tiles were hampered by slow rendering speeds.

> Luckily, through our team's many long hours of dedicated troubleshooting, these issues were resolved and new solutions were implemented. Nowadays, all the tiles sit amicably at the dinner table and do as they're told, which is to provide a seamless user experience for those who rely on aerial imagery and basemaps to do their jobs. The Discover services available today are much more robust than those from the old ArcGIS Image Server. It is cloud-based, utilizes Node.js and a Postgres database with all usage metrics stored in Google BigQuery, and is built to last.

Reviewer: ```suggestion
Luckily, through our team's many long hours of dedicated troubleshooting, these issues were resolved and new solutions were implemented. Nowadays, all the tiles sit amicably at the dinner table and do as they're told, which is to provide a seamless user experience for those who rely on aerial imagery and basemaps to do their jobs. The Discover services available today are much more robust than those from the old ArcGIS Image Server. Discover is cloud-based, utilizes Node.js and a Postgres database with all usage metrics stored in Google BigQuery, and is built to last.
```

## The Enduring Impact of Publicly Available Imagery

As of the writing of this article, Discover has seen over 5.4 billion requests over its lifetime. Each of these requests represents a pan or zoom made by a wide variety of users, from government agencies to engineering firms to data managers. These entities use the services for all kinds of workflows, including many that impact people in Utah every single day.

For example, the [Utah Division of Emergency Management](https://dem.utah.gov/) uses aerial imagery to safeguard Utah's future and help mitigate the effects of natural disasters in our state. In their words:

> 

Reviewer: These quotes may stand out more visually if we do block quotes (add a `>` at the beginning of each line). My first time reading through it kind of threw me for a loop when the voice changed, I didn't realize we were in a quote.

Reviewer 2: done

The [Utah Department of Transportation](https://www.udot.utah.gov/connect/) also uses aerial imagery in their daily work, helping every one of us get to where we need to go. Corey Unger, the spatial technologies manager at UDOT, had this to say about Discover:

“High quality aerial imagery, like that provided by UGRC is a valuable resource for various UDOT departments. It aids in fatal crash reviews, design processes, and environmental surveys. Designers utilize it for basemaps and asset location verification, while the LRS department relies on it for updating road information. The 6-inch resolution imagery is particularly preferred for its accuracy and detail, crucial for tasks like ADA ramp placement and road design.”

We are proud to provide the services for these incredible agencies, as well as many others. Throughout the state, all kinds of decision makers are using aerial imagery for:

- Water budgeting and distribution
- Rural public land issues
- 811 Call-Before-You-Dig center
- Broadband access expansion
- 911 and emergency response
- Natural resources management
- Infrastructure/Asset Management

The list could go on and on!

> The wide variety of use cases for Discover is truly what's worth celebrating. These services wouldn't be worth much if they just sat around collecting dust. But because of the amazing people we've had the joy of working with, those service become more than just pixels on a screen. Over the last ten years, we have had the opportunity to see these products transform lives and bolster data-driven decision making in our state. From the background processes that underpin everyday life, to the emergency management practices made possible through well-maintained data, Discover is just one way we “encourage and facilitate the effective use of geospatial information and technology for Utah.”

Reviewer: ```suggestion
The wide variety of use cases for Discover is truly what's worth celebrating. The basemaps and imagery wouldn't be worth much if they just sat around collecting dust. But, because of the amazing people we've had the joy of working with, those service become more than just pixels on a screen. Over the last ten years, we have had the opportunity to see these products transform lives and bolster data-driven decision making in our state. From the background processes that underpin everyday life, to the emergency management practices made possible through well-maintained data, Discover is just one way we “encourage and facilitate the effective use of geospatial information and technology for Utah.”
```
Trying to emphasize the content, not just the delivery platform.

## So, where did Discover come from?

Before we implemented the current Discover service, UGRC utilized a program called ArcGIS Image Server. It was available for public and private use, but connections made to the service created a heavy digital load on the [ArcGIS Server](https://enterprise.arcgis.com/en/image/latest/get-started/windows/what-is-arcgis-image-server-.htm). This resulted in a number of problems for both UGRC and our user base; the services experienced rendering latenecy and occasional downtime. While this old service did provide significant customization options for imagery and topographic maps, it was not the best solution.

For a period of time, UGRC resolved Utah's need for high-resolution aerial imagery through contracts with federal programs. These programs provided a patchwork of imagery every couple of years, but in 2012, those programs came to an end. Thus, we needed a new solution.

## From Giza to Discover

> Shortly after high resolution aerial imagery through federal programs was no longer available, UGRC began working with Google and [Sanborn](https://sanborn.com/) to create a tile pyrimid server appliance called Giza, which serves Google imagery and basemaps as WMS and WMTS. UGRC was one of the earliest adopters of this new Google imagery programs. We developed a Utah-specific version of Giza and named it Discover. In 2015, Discover officially launched as Utah's premier source for aerial imagery and basemaps services.

Reviewer 3: this new Google imagery programs.

This needs to go to these or programs needs to go to program.

When you view services through Discover, you may think you are looking at a one really big image of Utah. But, this giant photo of Utah is actually composed of hundreds of flight paths captured by airborne cameras, all mosaicked together into a series of 'tiles' (just like in a real mosaic floor or ceiling, it takes many colorful tiles to make up the whole artwork). These tiles are pre-cached, which allows for seamless transitions as you pan or zoom between them. They render so quickly, that many users don't even realize there is more than one tile to begin with.

However, like ragtag siblings stuck in their “get-along” shirt, some of these tiles didn't quite play nice at first. Some of them were so-called 'sticky tiles': when the user zoomed in or out, the other nearby tiles would zoom in or out as well, but sticky tiles would be stuck at the previous zoom level as though glued in place. Other problem children included 'delinquent tiles', which were tiles that didn't show up in the service at all. Apparently, those tiles had somewhere better to be. And many of the other perfectly innocent tiles were hampered by slow rendering speeds.

> Luckily, through our team's many long hours of dedicated troubleshooting, these issues were resolved and new solutions were implemented. Nowadays, all the tiles sit amicably at the dinner table and do as they're told, which is to provide a seamless user experience for those who rely on aerial imagery and basemaps to do their jobs. The Discover services available today are much more robust than those from the old ArcGIS Image Server. It is cloud-based, provides auto-scaling, and is built to last. AND WE NEED MORE FROM APPGEO HERE!

Reviewer 3: what kind of information do you want from app geo? I can give you more cloud service jargon if you want.

Reviewer 2: I tried for a couple weeks for them to get me a technical sentence from Calvin and all I got was `It's built using Node.js and a Postgres database. ` which works for me

The list could go on and on!

The wide variety of use cases for Discover is truly what's worth celebrating. These services wouldn't be worth much if they just sat around collecting dust. But because of the amazing people we've had the joy of working with, those service become more than just pixels on a screen. Over the last ten years, we have had the opportunity to see these products transform lives and bolster data-driven decision making in our state. From the background processes that underpin everyday life, to the emergency management practices made possible through well-maintained data, Discover is just one way we “encourage and facilitate the effective use of geospatial information and technology for Utah.”

> You can sign up for Discover and learn more about what we provide on our website. For further questions or inquiries, please reach out to <Contacts inline subject={frontmatter.title} contactKey='elevation' />.

Reviewer 3: this would benefit from a link to the discover sign up page.

Reviewer 2: done
